src/storage/txn/commands/flashback_to_version_read_phase.rs

```diff
@@ -60,46 +60,54 @@ impl<S: Snapshot> ReadCommand<S> for FlashbackToVersionReadPhase {
         }
         let mut reader = MvccReader::new_with_ctx(snapshot, Some(ScanMode::Forward), &self.ctx);
         // Scan the locks.
-        let (key_locks, has_remain_locks) = flashback_to_version_read_lock(
+        let (mut key_locks, has_remain_locks) = flashback_to_version_read_lock(
             &mut reader,
             &self.next_lock_key,
             &self.end_key,
             statistics,
         )?;
-        // Scan the writes.
-        let (mut key_old_writes, has_remain_writes) = flashback_to_version_read_write(
-            &mut reader,
-            key_locks.len(),
-            &self.next_write_key,
-            &self.end_key,
-            self.version,
-            self.start_ts,
-            self.commit_ts,
-            statistics,
-        )?;
+        // Scan the writes only if there is no more locks to unlock, otherwise, there
```

Comment:
how about separating into two stages, then the logic would be clearer, like 
```
match stage {
 ScanLock => xxx,
 ScanWrite => xxx,
}

Reply:
It has been changed to this, PTAL!